### PR TITLE
add task

### PR DESCRIPTION
**Pull request by AndriiYelieva**
[DEMO LINK](https://AndriiYelieva.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/components/NewTodo/NewTodo.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+type Props = {
+  onSubmit: (post: Todo) => void;
+};
+
+export const NewTodo: React.FC<Props> = ({ onSubmit }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleErorr, setHasTitleErorr] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasIserIdErorr, setHasIserIdErorr] = useState(false);
```
Comment (Viktor-Kost) [inline src/components/NewTodo/NewTodo.tsx:16]: Typo found in variable name 'hasIserIdErorr'. Consider renaming to 'hasUserIdError'.

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,47 @@
...
+}
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+
+  const onNewTodo = (todo: Todo) => {
+    const newTodo = {
+      ...todo,
+      completed: true,
+      id: getNewTodoId(todos),
```
Comment (Viktor-Kost) [inline src/App.tsx:29]: add this params on submit
```suggestion
```

File: src/components/NewTodo/NewTodo.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    setHasTitleErorr(!title);
+    setHasIserIdErorr(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      id: 0,
```
Comment (Viktor-Kost) [inline src/components/NewTodo/NewTodo.tsx:44]: use getNewTodoId to create id

File: src/components/UserInfo/UserInfo.tsx [hunk 4]
```diff
@@ -1 +1,14 @@
-export const UserInfo = () => {};
+import React from 'react';
+import { User } from '../../Types/User';
+
+type Props = {
+  user: User,
```
Comment (Viktor-Kost) [inline src/components/UserInfo/UserInfo.tsx:5]:
```suggestion
  user: User;
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Nice!
Some moments can be improved

File: src/App.tsx (hunk 2 above)
Comment (AndriiYelieva) [reply src/App.tsx:29]: I don't know how to do it

File: src/App.tsx [hunk 5]
```diff
@@ -1,61 +1,47 @@
...
+  const maxId = Math.max(...todos.map(todo => todo.id));
+
+  return maxId + 1;
+}
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+
+  const onNewTodo = (todo: Todo) => {
+    const newTodo = {
```
Comment (mykhalenych) [inline src/App.tsx:26]:
```suggestion
```

File: src/App.tsx [hunk 6]
```diff
@@ -1,61 +1,47 @@
...
+      ...todo,
+      completed: true,
+      id: getNewTodoId(todos),
+    };
+
+    setTodos((currentToDos) => ([
+      ...currentToDos,
+      newTodo,
```
Comment (mykhalenych) [inline src/App.tsx:34]:
```suggestion
      todo,
```

File: src/components/NewTodo/NewTodo.tsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+    setHasTitleErorr(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      id: 0,
```
Comment (mykhalenych) [inline src/components/NewTodo/NewTodo.tsx:44]:
```suggestion
      id: getNewTodoId(todos),
```

**Review by mykhalenych — CHANGES_REQUESTED** [review]

**Pull request closed without merging**